java-squid/src/main/java/org/sonar/java/se/SymbolicValue.java

```diff
@@ -181,9 +175,7 @@ public void computedFrom(List<SymbolicValue> symbolicValues) {
       for (ProgramState ps : result) {
         List<ProgramState> reversedStates = copyConstraint(rightOp, leftOp, ps, booleanConstraint);
         if (reversedStates.size() == 1 && reversedStates.get(0) == programState) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)